### ISSUE TITLE
Questions about the hyper-parameters.

### ISSUE THREAD

**Issue by D-X-Y** [Fezaries/CRnet#2]
Hi @Fezaries , thanks for your nice codes. In your paper, you use a different alpha and beta. Would you mind to let me know how do you tune these two parameters?

**Comment by Fezaries** [comment]
Hi @D-X-Y ,  in fact, the two modules shared the same coefficient for regularization terms when I first started building this network. And I tuned it and got a result which seemd not bad. But soon I realized that different coefficients should be used. So I fixed the coeffcient of one module and increased/decreased the coefficient of the other module to obseve the effect on the result, and finally obtained the values showed in the paper. 

I think this can be seen as a two-stage process:
A. Set alpha and beta as the same and tune them.
B. Fix one of them and tune the other.

**Comment by D-X-Y** [comment]
Thanks for your answer. I find that you use a different learning rate for your two modules. Tuning four hyper-parameters seems very difficult. How do you tune four hyper-parameters (2 LR, 2 weight decay).

**Comment by Fezaries** [comment]
Hi @D-X-Y , in the beginning, I took the network as a whole and only one LR and one weight decay are needed. In this case I obtained the LR with 1e-5 and the weight decay with 1e-5, which work well on most datasets and lead to ok results (AwA2: seen=0.8339, unseen=0.4656, h=0.5976. CUB: seen=0.5568, unseen=0.4246, h=0.4818). Based on these I attempted to tune the parameters in each modules separately for the sake of a higher accuracy and it truly made some effects. In this way tuning hyper-parameters is not so difficult and you can get a good result. 

Here are the traning logs with different hyper-parameter settings:
[(CUB LR=1E-5 WD=1E-5,1E-5)2019-08-27 21:55:08.log](https://github.com/Fezaries/CRnet/files/3547354/CUB.LR.1E-5.WD.1E-5.1E-5.2019-08-27.21.55.08.log)
[(AwA2 LR=1E-5 WD=1E-5,1E-5)2019-08-28 00:10:36.log](https://github.com/Fezaries/CRnet/files/3547356/AwA2.LR.1E-5.WD.1E-5.1E-5.2019-08-28.00.10.36.log)
[(AwA2 LR=1E-5 WD=1E-5,1E-4)2019-08-27 23:09:14.log](https://github.com/Fezaries/CRnet/files/3547357/AwA2.LR.1E-5.WD.1E-5.1E-4.2019-08-27.23.09.14.log)

**Comment by D-X-Y** [comment]
Cool, good strategy!

**Comment by D-X-Y** [comment]
@Fezaries Hi, would you mind to let me know the batch size for other datasets, (APY, SUN, CUB)?

**Comment by Fezaries** [comment]
@D-X-Y The batch size is fixed to 32 for all datasets.

**Comment by D-X-Y** [comment]
Thanks. How about the training episode and learning rate for C and R? Did you use different LR for those two networks in APY and SUN? @Fezaries

**Comment by D-X-Y** [comment]
Hi, @Fezaries would you mind to let me know the LR/weight-decay/training-epochs for APY and SUN?

**Comment by Fezaries** [comment]
@D-X-Y , sorry for my late reply. I uploaded the training code as well as models and logs for SUN and APY. Similar learning strategy is used. But it should be noted that the attributes of SUN and APY should be scaled to [0,100] (line 51) to keep consistant with AWA and CUB. And this may be the reason why you cannot get right results.

**Comment by D-X-Y** [comment]
@Fezaries I appreciate your help very much! Would you mind to provide some motivation on why scaling to 0~100? It is a little bit non-intuitive, because in most cases normalization could help improve performance.

**Comment by Fezaries** [comment]
@D-X-Y In fact, there isn't any motivation for training, but just for the contrast experiments. In the paper, I set the relation network (RN, "Learning to Compare: Relation Network for Few-Shot Learning") as a contrast. And RN directly uses the attributes of AWA and CUB for training, whose original values are from 0~100. So for consistency and comparison, our network takes the same input as RN.

**Comment by D-X-Y** [comment]
Thanks a lot!
